### PR TITLE
Run linter on build-system

### PR DESCRIPTION
**Pull request by rsimha-amp**
This PR adds ES5 compatible linter rules for the `build-system` directory, and ensures that any file in that directory that is touched by a PR is also linted. For now, we do this in warning mode, until a first round of fixes is checked in.

To run this locally, use:
```
gulp lint --build_system
```

Coming up: A follow up PR with slightly more sane lint rules, and an initial round of fixes for the `build-system` files.

Partial fix for #11273.

### CODE REVIEW

File: build-system/tasks/lint.js [hunk 1]
```diff
@@ -19,9 +19,11 @@
 var argv = require('minimist')(process.argv.slice(2));
 var config = require('../config');
 var eslint = require('gulp-eslint');
+var getStdout = require('../exec.js').getStdout;
```
Comment (erwinmombay) [inline build-system/tasks/lint.js]: remove .js

File: build-system/tasks/lint.js [hunk 2]
```diff
@@ -46,41 +68,79 @@ function isFixed(file) {
 }
 
 /**
- * Run the eslinter on the src javascript and log the output
- * @return {!Stream} Readable stream
+ * Initializes the linter stream based on globs
+ * @param {!object} globs
+ * @param {!object} streamOptions
```
Comment (erwinmombay) [inline build-system/tasks/lint.js]: `!Ojbect` instead

File: build-system/tasks/lint.js [hunk 3]
```diff
@@ -46,41 +68,79 @@ function isFixed(file) {
...
 
-  if (argv.fix) {
-    options.fix = true;
-  }
-
+/**
+ * Runs the linter on the given stream using the given options.
+ * @param {!string} path
+ * @param {!ReadableStream} stream
+ * @param {!object} options
```
Comment (erwinmombay) [inline build-system/tasks/lint.js]: `!Object`

**Comment by erwinmombay** [comment]
most of these wouldnt have been caught by the linter as they are bad type annotations

**Comment by rsimha-amp** [comment]
All your comments are addressed in the follow up PRs listed here.

File: build-system/tasks/lint.js (hunk 1 above)
Comment (rsimha-amp) [reply build-system/tasks/lint.js]: Done.

File: build-system/tasks/lint.js (hunk 2 above)
Comment (rsimha-amp) [reply build-system/tasks/lint.js]: Done.

File: build-system/tasks/lint.js (hunk 3 above)
Comment (rsimha-amp) [reply build-system/tasks/lint.js]: Done.

**Pull request merged**